### DISCUSSION TITLE
json pretty print

### DISCUSSION THREAD

**Discussion by GitHubsberg** [curl/curl#6767] [Ideas]
Add a --json-pretty-print option that pretty prints the response if the response is JSON. This saves the user from having to pipe the output to something that does the pretty printing. Often times you have to install something to do the pretty printing.

**Comment by dfandrich** [comment]
I think you'll find that a hard sell. With very few exceptions, curl
is agnostic to the output of a server. That matches the UNIX philosophy of
small tools that can be easily combined. It's just easy to pipe the output
to another tool to prettyprint JSON as it is to prettyprint XML, VCF, view PNG
or any other format. Moving those features into the curl tool isn't the right
place for them.

**Reply by bagder** [reply]
'jq' was pretty much made for exactly that purpose (and more)

**Comment by fossdd** [comment]
```
$ curl https://my.json | jq .
```
